ndk/src/input_queue.rs

```diff
@@ -33,45 +38,61 @@ impl InputQueue {
         self.ptr
     }
 
-    pub fn get_event(&self) -> Option<InputEvent> {
-        unsafe {
-            let mut out_event = ptr::null_mut();
-            if ffi::AInputQueue_getEvent(self.ptr.as_ptr(), &mut out_event) < 0 {
-                None
-            } else {
+    /// Returns the next available [`InputEvent`] from the queue.
+    ///
+    /// Returns [`std::io::ErrorKind::WouldBlock`] if no event is available.
```

Comment:
Should perhaps catch this case and return `Result<Option<>>` instead? That way they can easily write `while let Some(event) = get_event()` followed by either `?` or `.unwrap()`.

Reply:
it doesn't seem to be possible to differentiate between 'would block' and other errors, or is it? All error cases seem to map to `-1` /:

Reply:
Not sure I understand, the only error [this codepath] currently returns is `WOULD_BLOCK`, but there's no guarantee that it won't return other codes in the future. So for now it's either:

1. Yes there is an event;
2. No there is no event (`WouldBlock`);
3. Some other error that the user probably doesn't know/care about, abort.

[this codepath]: https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/jni/android_view_InputQueue.cpp;l=112;drc=7b3193e1e86ab5c0fcb784cb8616864045dd2515

Reply:
> All error cases seem to map to `-1` /:

In other words, @msiglreith what "map" are you referring to exactly?

Reply:
ah sorry, I don't remember where I looked but apparently at the wrong place 😅  - makes more sense now with the WOULD_BLOCK path

Reply:
@msiglreith :+1: - it's now changed to return `None` instead of putting the burden on the user to separate `e.kind() == WouldBlock` - which is a case you _must_ handle for this API to work/be useful (or call `has_events()` all the time?) - also implemented on `from_raw_os_error()` + `e.kind()` because there's no clean constant `-11` for `EWOULDBLOCK` in Rust.